Javadoc warnings

We are recently seeing a lot of Javadoc warnings in the travis build. we need to fix these.

```
19 warnings
[WARNING] Javadoc Warnings
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/config/DefaultProfileUtil.java:31: warning: no @return
[WARNING] public static String getDefaultActiveProfiles(){
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/config/DefaultProfileUtil.java:47: warning: no @param for app
[WARNING] public static void addDefaultProfile(SpringApplication app) {
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/service/LabelService.java:92: warning: no @param for pageable
[WARNING] public Page<Label> search(String query, Pageable pageable) {
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestInfiniteScrollEntityResource.java:160: warning: no @param for pageable
[WARNING] public ResponseEntity<List<FieldTestInfiniteScrollEntity>> searchFieldTestInfiniteScrollEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestInfiniteScrollEntityResource.java:160: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<FieldTestInfiniteScrollEntity>> searchFieldTestInfiniteScrollEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestPagerEntityResource.java:160: warning: no @param for pageable
[WARNING] public ResponseEntity<List<FieldTestPagerEntity>> searchFieldTestPagerEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestPagerEntityResource.java:160: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<FieldTestPagerEntity>> searchFieldTestPagerEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestPaginationEntityResource.java:160: warning: no @param for pageable
[WARNING] public ResponseEntity<List<FieldTestPaginationEntity>> searchFieldTestPaginationEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/FieldTestPaginationEntityResource.java:160: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<FieldTestPaginationEntity>> searchFieldTestPaginationEntities(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/LabelResource.java:153: warning: no @param for pageable
[WARNING] public ResponseEntity<List<Label>> searchLabels(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/LabelResource.java:153: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<Label>> searchLabels(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/OperationResource.java:160: warning: no @param for pageable
[WARNING] public ResponseEntity<List<Operation>> searchOperations(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/OperationResource.java:160: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<Operation>> searchOperations(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestInfiniteScrollResource.java:167: warning: no @param for pageable
[WARNING] public ResponseEntity<List<TestInfiniteScroll>> searchTestInfiniteScrolls(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestInfiniteScrollResource.java:167: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<TestInfiniteScroll>> searchTestInfiniteScrolls(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestPagerResource.java:167: warning: no @param for pageable
[WARNING] public ResponseEntity<List<TestPager>> searchTestPagers(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestPagerResource.java:167: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<TestPager>> searchTestPagers(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestPaginationResource.java:167: warning: no @param for pageable
[WARNING] public ResponseEntity<List<TestPagination>> searchTestPaginations(@RequestParam String query, Pageable pageable)
[WARNING] ^
[WARNING] /home/travis/app/src/main/java/com/mycompany/myapp/web/rest/TestPaginationResource.java:167: warning: no @throws for java.net.URISyntaxException
[WARNING] public ResponseEntity<List<TestPagination>> searchTestPaginations(@RequestParam String query, Pageable pageable)
[WARNING] ^
```

--- a/generators/entity/templates/src/main/java/package/service/_EntityService.java
+++ b/generators/entity/templates/src/main/java/package/service/_EntityService.java
@@ -16,7 +16,7 @@ public interface <%= entityClass %>Service {
 
     /**
      * Save a <%= entityInstance %>.
-     * 
+     *
      * @param <%= instanceName %> the entity to save
      * @return the persisted entity
      */
@@ -32,7 +32,7 @@ public interface <%= entityClass %>Service {
 <% for (idx in relationships) { if (relationships[idx].relationshipType == 'one-to-one' && relationships[idx].ownerSide != true) { -%>
     /**
      *  Get all the <%= entityInstancePlural %> where <%= relationships[idx].relationshipNameCapitalized %> is null.
-     *  
+     *
      *  @return the list of entities
      */
     List<<%= instanceType %>> findAllWhere<%= relationships[idx].relationshipNameCapitalized %>IsNull();
@@ -40,7 +40,7 @@ public interface <%= entityClass %>Service {
 
     /**
      *  Get the "id" <%= entityInstance %>.
-     *  
+     *
      *  @param id the id of the entity
      *  @return the entity
      */
@@ -48,15 +48,17 @@ public interface <%= entityClass %>Service {
 
     /**
      *  Delete the "id" <%= entityInstance %>.
-     *  
+     *
      *  @param id the id of the entity
      */
     void delete(<%= pkType %> id);<% if (searchEngine == 'elasticsearch') { %>
 
     /**
      * Search for the <%= entityInstance %> corresponding to the query.
-     * 
+     *
      *  @param query the query of the search
+     *  <% if (pagination != 'no') { %>
+     *  @param pageable the pagination information<% } %>
      *  @return the list of entities
      */
     <% if (pagination != 'no') { %>Page<<%= entityClass %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>);<% } %>

--- a/generators/entity/templates/src/main/java/package/web/rest/_EntityResource.java
+++ b/generators/entity/templates/src/main/java/package/web/rest/_EntityResource.java
@@ -139,8 +139,11 @@ public class <%= entityClass %>Resource {
      * SEARCH  /_search/<%= entityApiUrl %>?query=:query : search for the <%= entityInstance %> corresponding
      * to the query.
      *
-     * @param query the query of the <%= entityInstance %> search
-     * @return the result of the search
+     * @param query the query of the <%= entityInstance %> search <% if (pagination != 'no') { %>
+     * @param pageable the pagination information<% } if (fieldsContainNoOwnerOneToOne) { %>
+     * @param filter the filter of the request<% } %>
+     * @return the result of the search<% if (pagination != 'no') { %>
+     * @throws URISyntaxException if there is an error to generate the pagination HTTP headers<% } %>
      */
     @RequestMapping(value = "/_search/<%= entityApiUrl %>",
         method = RequestMethod.GET,

--- a/generators/entity/templates/src/main/java/package/web/rest/_EntityResource.java
+++ b/generators/entity/templates/src/main/java/package/web/rest/_EntityResource.java
@@ -140,8 +140,7 @@ public class <%= entityClass %>Resource {
      * to the query.
      *
      * @param query the query of the <%= entityInstance %> search <% if (pagination != 'no') { %>
-     * @param pageable the pagination information<% } if (fieldsContainNoOwnerOneToOne) { %>
-     * @param filter the filter of the request<% } %>
+     * @param pageable the pagination information<% } %>
      * @return the result of the search<% if (pagination != 'no') { %>
      * @throws URISyntaxException if there is an error to generate the pagination HTTP headers<% } %>
      */

--- a/generators/server/templates/src/main/java/package/config/_CacheConfiguration.java
+++ b/generators/server/templates/src/main/java/package/config/_CacheConfiguration.java
@@ -230,6 +230,8 @@ public class CacheConfiguration {
 
     /**
      * Use by Spring Security, to get events from Hazelcast.
+     *
+     * @return the session registry
      */
     @Bean
     public SessionRegistry sessionRegistry() {

--- a/generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java
+++ b/generators/server/templates/src/main/java/package/config/_DefaultProfileUtil.java
@@ -27,6 +27,8 @@ public final class DefaultProfileUtil {
 
     /**
      * Get a default profile from <code>application.yml</code>.
+     *
+     * @return the default active profile
      */
     public static String getDefaultActiveProfiles(){
         if (BUILD_PROPERTIES != null) {
@@ -43,6 +45,8 @@ public final class DefaultProfileUtil {
 
     /**
      * Set a default to use when no profile is configured.
+     *
+     * @param app the spring application
      */
     public static void addDefaultProfile(SpringApplication app) {
         Map<String, Object> defProperties =  new HashMap<>();
@@ -57,6 +61,8 @@ public final class DefaultProfileUtil {
 
     /**
      * Load application.yml from classpath.
+     *
+     * @return the YAML Properties
      */
     private static Properties readProperties() {
         try {

--- a/generators/server/templates/src/test/java/package/web/rest/_TestUtil.java
+++ b/generators/server/templates/src/test/java/package/web/rest/_TestUtil.java
@@ -51,6 +51,7 @@ public class TestUtil {
      *
      * @param size the size of the byte array
      * @param data the data to put in the byte array
+     * @return the JSON byte array
      */
     public static byte[] createByteArray(int size, String data) {
         byte[] byteArray = new byte[size];

